examples/coexist_klog_v1_and_v2/coexist_klog_v1_and_v2.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+
+	klogv1 "k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+// OutputCallDepth is the stack depth where we can find the origin of this call
+const OutputCallDepth = 6
+// log prefix that we have to strip out
+const DefaultPrefixLength = 53
+
+// klogWriter is used in SetOutputBySeverity call below to redirect
+// any calls to klogv1 to end up in klogv2
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
+	if len(p) < DefaultPrefixLength {
+		klogv2.InfoDepth(OutputCallDepth, string(p))
+		return len(p), nil
+	}
+	if p[0] == 'I' {
+		klogv2.InfoDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else if p[0] == 'W' {
+		klogv2.WarningDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else if p[0] == 'E' {
+		klogv2.ErrorDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else if p[0] == 'F' {
+		klogv2.FatalDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else {
+		klogv2.InfoDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	}
+	return len(p) - DefaultPrefixLength, nil
```

Comment:
we handled the full length, so return len(p)

From the writer interface:

> Write must return a non-nil error if it returns n < len(p).

Reply:
Fixed.